client/asset/eth/nodeclient.go

```diff
@@ -677,13 +460,6 @@ func (n *nodeClient) redeemToken(ctx context.Context, redemptions []ethv0.ETHSwa
 	// TODO: reject if there is duplicate secret hash
 	// TODO: use estimated gas
 	txOpts, _ := n.txOpts(ctx, 0, 300000, dexeth.GweiToWei(maxFeeRate))
-	n.nonceSendMtx.Lock()
-	defer n.nonceSendMtx.Unlock()
-	nonce, err := n.leth.ApiBackend.GetPoolNonce(ctx, n.creds.addr)
-	if err != nil {
-		return nil, fmt.Errorf("error getting nonce: %v", err)
-	}
-	txOpts.Nonce = new(big.Int).SetUint64(nonce)
 	if err := n.addSignerToOpts(txOpts); err != nil {
```

Comment:
I see some calls to `addSignerToOpts` left behind in several spots, but it looks like the intention is to have `(*nodeClient).txOpts` be doing that in addition to `GetPoolNonce`.